plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchange.java

```diff
@@ -298,8 +305,10 @@ public ExchangeSourceStatistics getExchangeSourceStatistics(ExchangeSourceHandle
     @Override
     public void close()
     {
+        ImmutableList.Builder<ListenableFuture<Void>> futures = ImmutableList.builder();
         for (Integer taskPartitionId : allSinks) {
-            exchangeStorage.deleteRecursively(getTaskOutputDirectory(taskPartitionId));
+            futures.add(exchangeStorage.deleteRecursively(getTaskOutputDirectory(taskPartitionId)));
         }
+        stats.getCloseExchange().record(Futures.allAsList(futures.build()));
```

Comment:
This is not on the critical path, do we need to record here too?

Reply:
It might be useful for tracking (for example if for some reason it starts to fail)